src/PhpSpec/CodeGenerator/Writer/TokenizedCodeWriter.php

```diff
@@ -58,6 +58,53 @@ public function insertAfterMethod(string $class, string $methodName, string $met
         return $this->insertStringAfterLine($class, $method, $line);
     }
 
+    public function insertImplementsInClass(string $class, string $interface) : string
```

Comment:
This needs cleaning up - the logic is hard to follow - I'd suggest extracting some private methods